modules/apps/29-fee/types/msgs.go

```diff
@@ -154,19 +149,29 @@ func (fee IdentifiedPacketFee) Validate() error {
 		return sdkerrors.Wrap(err, "failed to convert RefundAddress into sdk.AccAddress")
 	}
 
-	// if any of the fee's are invalid return an error
-	if !fee.Fee.AckFee.IsValid() || !fee.Fee.RecvFee.IsValid() || !fee.Fee.TimeoutFee.IsValid() {
-		return sdkerrors.ErrInvalidCoins
+	// enforce relayer is nil
+	if fee.Relayers != nil {
+		return ErrRelayersNotNil
 	}
 
-	// if all three fee's are zero or empty return an error
-	if fee.Fee.AckFee.IsZero() && fee.Fee.RecvFee.IsZero() && fee.Fee.TimeoutFee.IsZero() {
-		return sdkerrors.ErrInvalidCoins
+	// validate Fee
+	if err := fee.Fee.Validate(); err != nil {
+		return err
 	}
 
-	// enforce relayer is nil
-	if fee.Relayers != nil {
-		return ErrRelayersNotNil
+	return nil
+}
+
+// Validate asserts that each Fee is valid and all three Fees are not empty or zero
+func (fee Fee) Validate() error {
+	// if any of the fee's are invalid return an error
+	if !fee.AckFee.IsValid() || !fee.RecvFee.IsValid() || !fee.TimeoutFee.IsValid() {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidCoins, "contains one or more invalid fees")
+	}
+
+	// if all three fee's are zero or empty return an error
+	if fee.AckFee.IsZero() && fee.RecvFee.IsZero() && fee.TimeoutFee.IsZero() {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidCoins, "contains one or more invalid fees")
```

Comment:
Since all the fees are zero or empty.

```suggestion
		return sdkerrors.Wrap(sdkerrors.ErrInvalidCoins, "contains all invalid fees")
```

Reply:
I'd change the error message to be `all fees are zero`